app/views/shared/dossiers/editable_champs/_dossier_link.html.haml

```diff
@@ -5,7 +5,7 @@
     placeholder: "Numéro de dossier",
     autocomplete: 'off',
     required: champ.mandatory?,
-    data: { remote: true, url: champs_dossier_link_path(champ.id) }
+    data: { controller: 'input', action: 'input->input#onInput', input_url_value: champs_dossier_link_path(champ.id) }
```

Comment:
Pinaillage : je me demande si ici ça ne serait pas plus élégant de binder l'évènement "input" dans le controller stimulus  (au `connect`), plutôt que de l'écrire explicitement dans le DOM (ce qui est un peu verbeux).

À l'usage ça ressemblerait à ça :

```haml
    data: { controller: 'turbo-input', turbo_input_url_value: champs_dossier_link_path(champ.id) }
```

Mais comme tu le sens ; si l'explicite te semble mieux, bon pour moi 👍

Reply:
Alors je t'avoue que c'est le point sur lequel je suis encore un peu indécis avec `stimulus`. Quelle est la bonne pratique ? Le fait qu'il n'y a pas d'API declarative dans le controller pour le faire m’a fait penser que ce n’est pas "the right way”. Mais si ça te parait clair je suis chaud pour dire que si l’`event` est sur le `root element` on `bind` dans le `controller` et si c'est plus bas dans l'arborescence alors on le fait dans le template. Ça te va comme règle ?

Reply:
J'avoue j'ai pas d'idée générale non plus :) Mais la règle que tu propose m'a l'air bien, je pense qu'on peut partir là dessus 👍

Reply:
C'est fait. J'ai ajouté une méthode `on(eventName, handler)` sur stimulus application controller.